### PR TITLE
Fix a bug with projections and the aggregate_functions_null_for_empty setting (for query_plan_optimize_projection)

### PR DESCRIPTION
**Pull request by amosbird**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix a bug with projections and the aggregate_functions_null_for_empty setting. This was already fixed in PR #42198 but got forgotten after using query_plan_optimize_projection. This is for https://github.com/ClickHouse/ClickHouse/pull/49839 . 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 4764259f609a024f04ec8c082b63a495efc66d70 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49873/4764259f609a024f04ec8c082b63a495efc66d70/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Processors/QueryPlan/Optimizations/projectionsCommon.cpp [hunk 1]
```diff
@@ -42,6 +42,10 @@ bool canUseProjectionForReadingStep(ReadFromMergeTree * reading)
     if (reading->getContext()->getSettingsRef().allow_experimental_query_deduplication)
         return false;
 
+    // Currently projection don't support settings which implicitly modify aggregate functions.
+    if (reading->getContext()->getSettingsRef().aggregate_functions_null_for_empty)
+        return false;
+
```
Comment (CurtizJ) [inline src/Processors/QueryPlan/Optimizations/projectionsCommon.cpp:48]: Maybe also add this check to [old code](https://github.com/ClickHouse/ClickHouse/blob/25912a267311fc5aaa275edf6d72c2e1e0f97d34/src/Storages/MergeTree/MergeTreeData.cpp#L6606) for projections analysis?

**Comment by CurtizJ** [comment]
Failed tests:
- Stateless tests (debug, s3 storage) - https://github.com/ClickHouse/ClickHouse/issues/49185
- Stateless tests (tsan, s3 storage) - is also related to hedged requests therefore unrelated to this PR.

File: src/Processors/QueryPlan/Optimizations/projectionsCommon.cpp (hunk 1 above)
Comment (amosbird) [reply src/Processors/QueryPlan/Optimizations/projectionsCommon.cpp:48]: It's already there. See  #42198

Comment (CurtizJ) [reply src/Processors/QueryPlan/Optimizations/projectionsCommon.cpp:48]: Ah, sorry. Missed that.

**Pull request merged**